clients/src/test/java/org/apache/kafka/test/MockProducerInterceptor.java

```diff
@@ -86,5 +92,11 @@ public static void resetCounters() {
         ON_SUCCESS_COUNT.set(0);
         ON_ERROR_COUNT.set(0);
         ON_ERROR_WITH_METADATA_COUNT.set(0);
+        CLUSTER_META.set(null);
+    }
+
+    @Override
+    public void onUpdate(ClusterResource clusterResource) {
+        CLUSTER_META.set(clusterResource);
```

Comment:
Do you also want to verify that the cluster meta is set before `onAcknowledgement`?